trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/library/SvcompLibraryModel.java

```diff
@@ -0,0 +1,294 @@
+				loc, new CPointer(new CPrimitive(CPrimitives.CHAR)))));
+		result.add(new FunctionModel("__VERIFIER_nondet_charp", (main, node, loc, name) -> handleVerifierNonDet(main,
+				loc, new CPointer(new CPrimitive(CPrimitives.CHAR)))));
+		result.add(new FunctionModel("__VERIFIER_nondet_float",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.FLOAT))));
+		result.add(new FunctionModel("__VERIFIER_nondet_double",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.DOUBLE))));
+		result.add(new FunctionModel("__VERIFIER_nondet_int",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.INT))));
+		result.add(new FunctionModel("__VERIFIER_nondet_long",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.LONG))));
+		result.add(new FunctionModel("__VERIFIER_nondet_longlong",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.LONGLONG))));
+		result.add(new FunctionModel("__VERIFIER_nondet_int128",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.INT128))));
+		result.add(new FunctionModel("__VERIFIER_nondet_short",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.SHORT))));
+		result.add(new FunctionModel("__VERIFIER_nondet_uchar",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.UCHAR))));
+		result.add(new FunctionModel("__VERIFIER_nondet_unsigned_char",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.UCHAR))));
+		result.add(new FunctionModel("__VERIFIER_nondet_unsigned",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.UINT))));
+		result.add(new FunctionModel("__VERIFIER_nondet_unsigned_int",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.UINT))));
+		result.add(new FunctionModel("__VERIFIER_nondet_uint",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.UINT))));
+		result.add(new FunctionModel("__VERIFIER_nondet_ulong",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.ULONG))));
+		result.add(new FunctionModel("__VERIFIER_nondet_ulonglong",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.ULONGLONG))));
+		result.add(new FunctionModel("__VERIFIER_nondet_uint128",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.UINT128))));
+		result.add(new FunctionModel("__VERIFIER_nondet_ushort",
+				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.USHORT))));
+
+		// TODO: These are no predefined types, thus the return value may depend on the benchmark
```

Comment:
We should get rid of such special handling. If there are still benchmarks with these, either they have to be changed, or the newly proposed extension of `__VERIFIER_nondet` to more types should cover them.

(Not an issue for this PR though.)